### PR TITLE
DSP-1033 Reposition List Nodes

### PR DESCRIPTION
**Pull request by SepidehAlassi**
resolves DSP-1033
new route for this: PUT `admin/lists/<nodeIri>/position`. Requires `parentIri` and the new `position` in the request body.
- [x] reposition a child node within the same parent, shift siblings accordingly.
- [x] change parent of a child node, and position it in a specific position among new siblings. Shift the new siblings accordingly.
- [x] if position=-1, move the node to the end of children of  the given parent node.
- [x] add tests 
- [x] add test data
- [x] add documentation

### CODE REVIEW

File: webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponderADM.scala [hunk 1]
```diff
@@ -1181,6 +1182,276 @@ class ListsResponderADM(responderData: ResponderData) extends Responder(responde
...
+         * @param isNewParent identifier that node is added to another parent or not.
+         *
+         * @throws BadRequestException if given position is out of range.
+         */
+        def isNewPositionValid(parentNode: ListNodeADM, isNewParent: Boolean): Unit = {
+            val numberOfChildren = parentNode.getChildren.size
+            // If the node must be added to a new parent, highest valid position is numberOfChildren+1
+            // That means the furthests a node can be positioned is being appended to the end of children of the new parent.
+            if(isNewParent && changeNodePositionRequest.position >  numberOfChildren + 1) {
+                throw BadRequestException(s"Invalid position given, maximum allowed is=${numberOfChildren + 1}!")
```
Comment (benjamingeer) [inline webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponderADM.scala:1217]: Maybe this is just me, but I don't like receiving error messages with exclamation marks in them, it makes me feel like the computer is shouting at me. :)

File: webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponderADM.scala [hunk 2]
```diff
@@ -1181,6 +1182,276 @@ class ListsResponderADM(responderData: ResponderData) extends Responder(responde
...
+            }
+
+            // If node remains in its current parent, the highest valid position is numberOfChildren
+            // That means nodes are only reorganized within the same parent.
+            if (!isNewParent && changeNodePositionRequest.position > numberOfChildren) {
+                throw BadRequestException(s"Invalid position given, maximum allowed is=${numberOfChildren}!")
+            }
+        }
+        /**
+         * Checks that the position of the node is updated and node is sublist of specified parent.
```
Comment (benjamingeer) [inline webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponderADM.scala:1227]: Please finish the `@throws` comment.

**Review by benjamingeer — APPROVED** [review]
This looks great. Please try reformatting these files using the [dsp-intellij-settings](https://github.com/dasch-swiss/dsp-intellij-settings) repo that I've just made.

**Comment by benjamingeer** [comment]
Did I review this by mistake? I thought I had got a review request but maybe I didn't. :)

**Review by benjamingeer — CHANGES_REQUESTED** [review]
This is even better than the first time I reviewed it. :) But please reformat it with the shared IntelliJ settings.

**Comment by SepidehAlassi** [comment]
> This is even better than the first time I reviewed it. :) But please reformat it with the shared IntelliJ settings.

I have reformatted it already :/

**Comment by benjamingeer** [comment]
> I have reformatted it already :/

I've just done commit b056bb2 so you can see what `Reformat Code` does in my IntelliJ with that config.

**Comment by benjamingeer** [comment]
Hmm but I don't like what it did here:

```scala
            sparqlQuery <- Future(org.knora.webapi.messages.twirl.queries.sparql.admin.txt.getListNode(
                triplestore = settings.triplestoreType,
                nodeIri = nodeIri
            ).toString())
            ).toString()
            )
```

I wanted it to look like this:

```scala
            sparqlQuery <- Future(
                org.knora.webapi.messages.twirl.queries.sparql.admin.txt.getListNode(
                    triplestore = settings.triplestoreType,
                    nodeIri = nodeIri
                ).toString()
            )
```

**Review by benjamingeer — APPROVED** [review]
I've fixed the formatting in `ListsResponderADM` manually, and changed the settings repository to hopefully produce better results.

**Comment by SepidehAlassi** [comment]
> I've fixed the formatting in `ListsResponderADM` manually, and changed the settings repository to hopefully produce better results.

thank you very much for reformatting the code and for reviewing this PR.

**Comment by subotic** [comment]
@SepidehAlassi @benjamingeer I think that it should be possible to store all settings inside https://github.com/dasch-swiss/dsp-api/blob/main/.editorconfig. The EditorConfig plugin comes bundled with IntelliJ. This would also allow, other IDEs to be configured correctly.

**Review by subotic — APPROVED** [review]
Thanks, looks great!

**Pull request merged**